### PR TITLE
db refactoring and change password interface

### PR DESCRIPTION
**Pull request by karenetheridge**
- more DBIC usage - Conch::Model::User and Conch::Model::SessionToken are now fully converted
- db_* helper methods added, for easy access to db queries
- LP-46272723: user can change their own password

### CODE REVIEW

File: lib/Conch/Controller/User.pm [hunk 1]
```diff
@@ -134,16 +149,19 @@ Get the individual key/value pair for a setting for the User
...
+		{ name => $key },
+		{ order_by => { -desc => 'created' } },
+	)->first;
 
 	return $c->status( 404, { error => "No such setting '$key'" } )
-		unless $settings->{$key};
+		unless $setting;
 
-	$c->status( 200, { $key => $settings->{$key} } );
+	$c->status( 200, { $key => from_json($setting->value) } );
```
Comment (sungo) [inline lib/Conch/Controller/User.pm]: the 'status' helper should be handling the serialization for you at https://github.com/joyent/conch/blob/master/lib/Conch.pm#L109

*should* is operative, I think.

File: lib/Conch/Controller/User.pm [hunk 2]
```diff
@@ -155,21 +173,48 @@ Delete a single setting for a user, provided it was set previously
...
+Stores a new password for the current user.
+
+Optionally takes a query parameter 'clear_tokens' (defaulting to true), to also revoke session
+tokens for the user, forcing all tools to log in again.
+
+=cut
+
+sub change_password ($c) {
+	my $body = $c->req->json;
+	return $c->status( 400, { error => 'Payload required' } ) unless $body;
```
Comment (sungo) [inline lib/Conch/Controller/User.pm]: Let's move this to use the json validator like https://github.com/joyent/conch/blob/master/lib/Conch/Controller/DB/HardwareProduct.pm#L84

File: lib/Conch/Controller/User.pm (hunk 1 above)
Comment (karenetheridge) [reply lib/Conch/Controller/User.pm]: this is deserialization, from the db.

Comment (sungo) [reply lib/Conch/Controller/User.pm]: Roger

File: lib/Conch/Controller/User.pm (hunk 2 above)
Comment (karenetheridge) [reply lib/Conch/Controller/User.pm]:
```
    #          $got->{TEST} = '"TEST"'
    #     $expected->{TEST} = 'TEST'
```

Comment (karenetheridge) [reply lib/Conch/Controller/User.pm]: ok, I'm adding an entry to json-schema/input.yaml, and then this to `change_password`:

```
 sub change_password ($c) {
-       my $body = $c->req->json;
-       return $c->status( 400, { error => 'Payload required' } ) unless $body;
+       my $body =  $c->validate_input('UserPassword');
+       return if not $body;
 
        my $new_password = $body->{password};
-       return $c->status(400, { error => 'Missing parameter "password"' }) unless $new_password;
 
```

**Pull request merged**